meshio/gmsh/_gmsh41.py

```diff
@@ -155,13 +184,23 @@ def _read_nodes(f, is_ascii, data_size):
         # Store the point densely and in the order in which they appear in the file.
         # x(double) y(double) z(double) (* numNodes)
         points[ixx] = fromfile(f, c_double, num_nodes * 3).reshape((num_nodes, 3))
+
+        # Entity tag and entity dimension of the nodes. Stored as point-data.
+        entity_of_nodes[ixx] = entity_tag
+        entity_dimension[ixx] = dim
         idx += num_nodes
 
     _fast_forward_to_end_block(f, "Nodes")
-    return points, tags
 
+    # Entity information for nodes is stored as point data, as an nx2 array
+    dim_tags = numpy.vstack((entity_dimension, entity_of_nodes)).T
```

Comment:
Even easier: `numpy.array([entity_dimension, entity_dimension])`. This way, the data remains contiguous in memory, too. And even better: Instead of `entity_of_nodes` and `entity_dimension`, instantiate an n-by-2 array `dim_tags` right at the start.

Reply:
Agree. pushed.